tidy3d/components/geometry.py

```diff
@@ -980,6 +1034,73 @@ def set_center(cls, val, values):
         values["center"] = cls.unpop_axis(z0, (x0, y0), axis=values.get("axis"))
         return val
 
+    @pydantic.validator("vertices", always=True)
+    def no_crossing(cls, val, values):
```

Comment:
I think If this is slow, we can simply return early if dilation and sidewall are not defined, otherwise it's fine to check always.